### PR TITLE
Introduce inclusive naming check and update existing language

### PR DESCRIPTION
**Pull request by aaronshurley**
We are following the Inclusive Naming Initiative's guidance as that is what the CNCF supports.

#### What this PR does / why we need it:

This PR addresses https://github.com/vmware-tanzu/carvel/issues/594

#### Which issue(s) this PR fixes:
Fixes https://github.com/vmware-tanzu/carvel/issues/594

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

##### Review Checklist:

- [x] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [x] Relevant tests are added or updated
- [x] Relevant docs in this repo added or updated
- [x] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [x] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by aaronshurley** [comment]
created as a draft until https://github.com/vmware-tanzu/carvel/pull/595 lands

File: examples/cf-for-k8s-v0.2.0-custom/config.yml
Comment (praveenrewar) [inline examples/cf-for-k8s-v0.2.0-custom/config.yml:10780]: Maybe we should include the complete examples directory in the wokeignorefile? (As almost all the the config files are generated through some instructions and we don't have much control over it)

I am fine with this approach as well :)

**Review by praveenrewar — APPROVED** [review]
LGTM!

(Other than one comment which we can choose to skip as well)

File: examples/cf-for-k8s-v0.2.0-custom/config.yml
Comment (aaronshurley) [reply examples/cf-for-k8s-v0.2.0-custom/config.yml:10780]: @praveenrewar Good question. I don't have a strong opinion on this. Since you're a primary maintainer of this repo, I'll leave it to you to decide on the long-term plan. I'm going to merge this PR as is. Thanks for the review!

**Pull request merged**
